packages/flutter/lib/src/material/dialog.dart

```diff
@@ -55,7 +61,26 @@ class Dialog extends StatelessWidget {
     this.alignment,
     this.child,
   }) : assert(clipBehavior != null),
-       assert(elevation == null || elevation >= 0.0);
+       assert(elevation == null || elevation >= 0.0),
+       _fullscreen = false;
+
+  /// Creates a fullscreen dialog.
+  ///
+  /// Typically used in conjunction with [showDialog].
+  const Dialog.fullscreen({
+    super.key,
+    this.backgroundColor,
+    this.child,
+  }) : elevation = 0,
+       shadowColor = null,
+       surfaceTintColor = null,
+       insetAnimationDuration = const Duration(milliseconds: 100),
```

Comment:
Where does this duration come from? Also I would see someone perhaps wanting to customize this and other properties. Can we ensure the same set of customizeable properties are available for `Dialog` and `Dialog.fullscreen`?

Reply:
That's a great point.  I was conservative with this since it's much easier to add parameters than to remove them. 

I did some native testing, inset animation is instant, as you can see below:

https://user-images.githubusercontent.com/48603081/192486059-de3e0e3b-67e4-46c3-b691-dfa362165b43.mp4

>  Also I would see someone perhaps wanting to customize this 

I am setting `insetAnimationDuration` to  `Duration.zero` default for fullscreen dialog here, with the ability to customize.